### PR TITLE
Model factory

### LINKED ISSUE #9: implement ModelDesign Facotry
- json schema endpoint
- json v6 validation ?Ajv (https://github.com/epoberezkin/ajv)

### PR DESCRIPTION
**Pull request by edoardocavazza**
This PR introduces the `model` factory, as well as a lot of new models.
Closes #9

**Features**
* New `model` factory
* `properties`, `property_types` and `object_types` models and collections

**Changes**
* replace `registry` factory with the new `model` one
* add a `defaultEndpoint` getter to the Collection in order to handle not type-based endpoints

### CODE REVIEW

File: src/factories/model.js [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+                collection.findAll()
+                    .then(() => collection)
+            );
+    }
+
+    /**
+     * Fecth abstract object types.
+     * @return {Promise<ObjectTypesCollection>}
+     */
+    getAbastractObjectTypes() {
```
Comment (didoda) [inline src/factories/model.js]: typo: `getAbstractObjectTypes`

File: src/factories/model.js [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+    getObjectTypes() {
+        return this.initClass(ObjectTypesCollection)
+            .then((collection) =>
+                collection.findAll()
+                    .then(() => collection)
+            );
+    }
+
+    /**
+     * Fecth abstract object types.
```
Comment (didoda) [inline src/factories/model.js]: typo: `Fetch`

File: src/factories/model.js [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+    }
+
+    /**
+     * Get the ObjectType Model for a given type.
+     * @param {String} type The type of the Model.
+     * @return {Promise<ObjectType>}
+     */
+    getObjectType(type) {
+        return this.initClass(ObjectTypesCollection)
+            .then((collection) => collection.findById(type));
```
Comment (didoda) [inline src/factories/model.js]: `findById(type)` is right? I'd expect and `id` as parameter to `findById`

Comment (edoardocavazza) [reply src/factories/model.js]: In this case, the type name is a string identifier.

**Pull request merged**